gufe/tests/test_transformation.py

```diff
@@ -0,0 +1,114 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+import pytest
+
+from gufe.transformations import Transformation, NonTransformation
+
+from .test_protocol import DummyProtocol, DummyProtocolResult
+
+
+@pytest.fixture
+def absolute_transformation(solvated_ligand, solvated_complex):
+    return Transformation(
+        solvated_ligand,
+        solvated_complex,
+        protocol=DummyProtocol(settings=None),
+        mapping=None,
+    )
+
+
+@pytest.fixture
+def complex_equilibrium(solvated_complex):
+    return NonTransformation(solvated_complex, protocol=DummyProtocol(settings=None))
+
+
+class TestTransformation:
+    def test_init(self, absolute_transformation, solvated_ligand, solvated_complex):
+        tnf = absolute_transformation
+
+        assert tnf.initial is solvated_ligand
+        assert tnf.final is solvated_complex
+
+    def test_protocol(self, absolute_transformation):
+        tnf = absolute_transformation
+
+        assert isinstance(tnf.protocol, DummyProtocol)
+
+        protocoldag = tnf.create()
+        protocoldagresult = protocoldag.execute()
+
+        protocolresult = tnf.gather([protocoldagresult])
+
+        assert isinstance(protocolresult, DummyProtocolResult)
+
+        len(protocolresult.data) == 1
+
+    def test_equality(self, absolute_transformation, solvated_ligand, solvated_complex):
+
+        opposite = Transformation(
+            solvated_complex, solvated_ligand, protocol=DummyProtocol(settings=None)
+        )
+        assert absolute_transformation != opposite
+
+        different_protocol_settings = Transformation(
+            solvated_ligand,
+            solvated_complex,
+            protocol=DummyProtocol(settings={"lol": True}),
+        )
+        assert absolute_transformation != different_protocol_settings
+
+        identical = Transformation(
+            solvated_ligand,
+            solvated_complex,
+            protocol=DummyProtocol(settings=None),
+            mapping=None,
+        )
+        assert absolute_transformation == identical
+
+    def test_dict_roundtrip(self):
+        # TODO: need registration of `Protocol`s for this to work
+        ...
+
+
+class TestNonTransformation:
```

Comment:
From a cursory look these two classes seem like they could be squashed to the one base class with just ntnf renamed to something that's declared as a class attribute?

Reply:
Yes, there is some potential for consolidation, but there are some differences in `test_equality`.